### ISSUE TITLE
footnote support

### ISSUE THREAD

**Issue by cittadhammo** [LegNeato/mdbook-typst#10]
Would it be possible to add footnote support:
```

Here is a simple footnote[^1]. With some additional text after it.

[^1]: My reference.
```

**Comment by LegNeato** [comment]
I think it would be easy from the markdown side, but what should they be converted to on the mdbook side? I don't think mdbook supports footnotes but perhaps I am mistaken.

**Comment by cittadhammo** [comment]
Yes sorry, there is an explanation here for example https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/

or here https://rust-lang.github.io/mdBook/format/markdown.html#:~:text=Footnotes,a%20caret%20at%20the%20front.
for mdbook proper

My workflow is to write a book. So far, I have used a mdbook to latex converter, but when I saw your project, I thought I could give it a try. It works very wll, the only thing missing is that the footnote should be nicely integrated on the footer of the page where they appear, Latex is very good at this.

What do you think?

**Comment by cittadhammo** [comment]
@LegNeato Good, thank you, I have never coded in rust, but I will have a look at your code and why not do a PR if I can ;-)

**Comment by LegNeato** [comment]
Working on this, should have it done in the next week

**Comment by LegNeato** [comment]
So the annoying part of doing this is we'll have to buffer the entire document to support these, as the footnote links can be anywhere in the document.

**Comment by cittadhammo** [comment]
Yes, this is why I added this other function to store the references of the footnotes: https://github.com/LegNeato/mdbook-typst/pull/16/commits/d1aa4ba4eefac7557ad5e48f1d1ed2b04827dcea

it is still very fast, even for my 300+ pages project.

**Comment by michalfita** [comment]
@LegNeato but today the the Markdown with footnote leads to the following:
```
error: unexpected argument
    ┌─ book.typst:245:153
    │
245 │ #par()[Not long ago that was quite a challenge, but today it's easy and accessible - use diagrams where you can. #emph[A picture is worth thousand words][^brisbane] in technical details, graphs and diagrams illustrate all form of relationships much better than written word (with some exceptions around UML, but it's not important here). Diagrams of relationships in code of your product(s) matter a lot for people having just first chance to look into it, your new joiners are going to bless their presence during on-boarding process.]
    │                                                                                                                                                          ^^^^^^^^^^^

2025-07-15 11:53:14 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.
2025-07-15 11:53:14 [ERROR] (mdbook::utils): Error: Rendering failed
2025-07-15 11:53:14 [ERROR] (mdbook::utils): 	Caused By: The "typst" renderer failed
```
Looks like the `[^brisbane]` is not caught and send for rendering w/o escaping, something `typst` itself doesn't understand.

**Comment by cittadhammo** [comment]
I think you need to remove the italic. It creates confusion for the parser.

<img width="352" height="90" alt="Image" src="https://github.com/user-attachments/assets/c925e3cd-36bc-4921-9558-2f732c46db68" />

PS: Sorry, I closed the issue and I cannot reopen.

**Comment by michalfita** [comment]
But that's correct Markdown that renders correctly to both HTML and EPUB and to PDF with `mdbook-pandoc` extension.

**Comment by cittadhammo** [comment]
Yes. I think it is a bug in the function I made in the converter. Not sure.
There is also a lot of problems if there are italic in the footnote content.

The function I made is very basic and you can debug by uncommenting in the [converter.rs](https://github.com/LegNeato/mdbook-typst/commit/d1aa4ba4eefac7557ad5e48f1d1ed2b04827dcea)

<img width="1500" height="184" alt="Image" src="https://github.com/user-attachments/assets/e38421f3-b6c6-4ebe-94a3-7f6aa4ce3982" />
